### ISSUE TITLE
Multiple images in the same Helm chart, argocd image updater won't update correctly 

### ISSUE THREAD

**Issue by jeff51419** [argoproj-labs/argocd-image-updater#202]
**Describe the bug**
When I try to use Multiple images in the same Helm chart, image tag will auto rollback or update.
it won't fix to specific tag.  

**To Reproduce**
1. kind create cluster
2. kubectl create namespace argocd
3. kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj/argo-cd/v2.0.0-rc1/manifests/install.yaml
4. ssh-keygen -t ed25519 -C argocd-image-updater-lab -f ~/argocd-image-updater-lab_ed25519
5. argocd repo add git@github.com:jeff51419/argocd-image-updater-example.git --name ......
6. kubectl apply -n argocd -f https://raw.githubusercontent.com/argoproj-labs/argocd-image-updater/v0.9.0/manifests/install.yaml
7. argocd app create -f lab.yaml
```yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: helm-guestbook
spec:
  destination:
    namespace: default
    server: https://kubernetes.default.svc
  project: default
  source:
    helm:
      valueFiles:
        - values.yaml
    path: helm-guestbook
    repoURL: git@github.com:jeff51419/argocd-image-updater-example.git
    targetRevision: main
  syncPolicy:
    automated: {}
```
![image](https://user-images.githubusercontent.com/18411883/116852608-38b4b100-ac27-11eb-9778-c115d25c81ad.png)

8. add annotate
```
kubectl annotate application helm-guestbook \
argocd-image-updater.argoproj.io/write-back-method=git \
argocd-image-updater.argoproj.io/image-list="guestbook=gcr.io/heptio-images/ks-guestbook-demo, nginx=nginx" \
argocd-image-updater.argoproj.io/guestbook.helm.image-name=guestbook.image \
argocd-image-updater.argoproj.io/guestbook.helm.image-tag=guestbook.tag \
argocd-image-updater.argoproj.io/nginx.helm.image-name=nginx.image \
argocd-image-updater.argoproj.io/nginx.helm.image-tag=nginx.tag
```
![image](https://user-images.githubusercontent.com/18411883/116852632-410cec00-ac27-11eb-8b5b-8d78f7827088.png)

9. check ***.argocd-source-helm-guestbook.yaml***
![image](https://user-images.githubusercontent.com/18411883/116852721-6994e600-ac27-11eb-82a5-af4dc103937c.png)

10. wait for 10 mins, and we can notice it keep change.
![image](https://user-images.githubusercontent.com/18411883/116852808-93e6a380-ac27-11eb-87fc-01758d5f0afa.png)

Sometime it will remove guestbook image block in ***.argocd-source-helm-guestbook.yaml***
https://github.com/jeff51419/argocd-image-updater-example/commit/33b680af20cf653914705de04cdbe37d9e38a6d5
![image](https://user-images.githubusercontent.com/18411883/116853130-1cfdda80-ac28-11eb-8940-83e83e1e18db.png)

some it will remove nginx image block in ***.argocd-source-helm-guestbook.yaml***
![image](https://user-images.githubusercontent.com/18411883/116853226-4c144c00-ac28-11eb-880d-be8ce525fef0.png)

these will affect the image tag will rollback to *value.yaml* value
![image](https://user-images.githubusercontent.com/18411883/116853444-a1e8f400-ac28-11eb-822b-dc0e12102931.png)

**Expected behavior**
I expected the ***.argocd-source-helm-guestbook.yaml*** should be as same as the first time argocd-image-updater commit.
And it will fixed to the specific tag of images.
```yaml
helm:
  parameters:
  - name: guestbook.image
    value: gcr.io/heptio-images/ks-guestbook-demo
    forcestring: true
  - name: guestbook.tag
    value: "0.2"
    forcestring: true
  - name: nginx.image
    value: nginx
    forcestring: true
  - name: nginx.tag
    value: "1.20"
    forcestring: true
```

**Additional context**
![image](https://user-images.githubusercontent.com/18411883/116854306-11abae80-ac2a-11eb-9496-b78f5ff62db3.png)

guestbook-deployment.yaml
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: guestbook
spec:
  selector:
    matchLabels:
      app: guestbook
  replicas: 1 
  template:
    metadata:
      labels:
        app: guestbook
    spec:
      containers:
      - name: guestbook
        image: "{{ .Values.guestbook.image }}:{{ .Values.guestbook.tag }}"
        imagePullPolicy: {{ .Values.guestbook.pullPolicy }}
        ports:
        - containerPort: 80
```

nginx-deployment.yaml
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: nginx
spec:
  selector:
    matchLabels:
      app: nginx
  replicas: 1 
  template:
    metadata:
      labels:
        app: nginx
    spec:
      containers:
      - name: nginx
        image: "{{ .Values.nginx.image }}:{{ .Values.nginx.tag }}"
        imagePullPolicy: {{ .Values.nginx.pullPolicy }}
        ports:
        - containerPort: 80
```
source: https://github.com/jeff51419/argocd-image-updater-example

**Version**
v0.9.0

**Logs**
```
time="2021-05-03T08:14:14Z" level=info msg="Successfully updated image 'gcr.io/heptio-images/ks-guestbook-demo:0.1' to 'gcr.io/heptio-images/ks-guestbook-demo:0.2', but pending
  spec update (dry run=false)" alias=guestbook application=helm-guestbook image_name=heptio-images/ks-guestbook-demo image_tag=0.1 registry=gcr.io
 time="2021-05-03T08:14:18Z" level=info msg="Starting configmap/secret informers"
 time="2021-05-03T08:14:18Z" level=info msg="Setting new image to nginx:1.20.0" alias=nginx application=helm-guestbook image_name=nginx image_tag=1.20 registry=
 time="2021-05-03T08:14:18Z" level=info msg="Successfully updated image 'nginx:1.20' to 'nginx:1.20.0', but pending spec update (dry run=false)" alias=nginx application=helm-gue
 stbook image_name=nginx image_tag=1.20 registry=
 time="2021-05-03T08:14:18Z" level=info msg="Committing 2 parameter update(s) for application helm-guestbook" application=helm-guestbook
 time="2021-05-03T08:14:19Z" level=info msg="Configmap/secret informer synced"
 time="2021-05-03T08:14:19Z" level=info msg="secrets informer cancelled"
 time="2021-05-03T08:14:19Z" level=info msg="configmap informer cancelled"
 time="2021-05-03T08:14:19Z" level=info msg="rm -rf /tmp/git-helm-guestbook306988303" dir= execID=lvbTj
 time="2021-05-03T08:14:19Z" level=info msg="Initializing git@github.com:jeff51419/argocd-image-updater-example.git to /tmp/git-helm-guestbook306988303"
 time="2021-05-03T08:14:19Z" level=warning msg="temporarily disabling strict host key checking (i.e. '-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null'), please don't
  use in production"
 time="2021-05-03T08:14:19Z" level=info msg="git fetch origin --tags --force" dir=/tmp/git-helm-guestbook306988303 execID=GI5nn
 time="2021-05-03T08:14:21Z" level=info msg="git config user.name argocd-image-updater" dir=/tmp/git-helm-guestbook306988303 execID=SWBYf
 time="2021-05-03T08:14:21Z" level=info msg="git config user.email noreply@argoproj.io" dir=/tmp/git-helm-guestbook306988303 execID=t7nUZ
 time="2021-05-03T08:14:21Z" level=info msg="git checkout --force main" dir=/tmp/git-helm-guestbook306988303 execID=Y3NAO
 time="2021-05-03T08:14:21Z" level=info msg="git clean -fdx" dir=/tmp/git-helm-guestbook306988303 execID=I011m
 time="2021-05-03T08:14:21Z" level=info msg="git commit -a -m Update to new image versions" dir=/tmp/git-helm-guestbook306988303 execID=p5XvB
 time="2021-05-03T08:14:21Z" level=warning msg="temporarily disabling strict host key checking (i.e. '-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null'), please don't
  use in production"
 time="2021-05-03T08:14:21Z" level=info msg="git push origin main" dir=/tmp/git-helm-guestbook306988303 execID=uaBEg
 time="2021-05-03T08:14:24Z" level=info msg="Successfully updated the live application spec" application=helm-guestbook
 time="2021-05-03T08:14:24Z" level=info msg="Processing results: applications=1 images_considered=2 images_skipped=0 images_updated=2 errors=0"
 time="2021-05-03T08:16:24Z" level=info msg="Starting image update cycle, considering 1 annotated application(s) for update"
 time="2021-05-03T08:16:25Z" level=info msg="Setting new image to gcr.io/heptio-images/ks-guestbook-demo:0.2" alias=guestbook application=helm-guestbook image_name=heptio-images
 /ks-guestbook-demo image_tag=0.1 registry=gcr.io
 time="2021-05-03T08:16:25Z" level=info msg="Successfully updated image 'gcr.io/heptio-images/ks-guestbook-demo:0.1' to 'gcr.io/heptio-images/ks-guestbook-demo:0.2', but pending
  spec update (dry run=false)" alias=guestbook application=helm-guestbook image_name=heptio-images/ks-guestbook-demo image_tag=0.1 registry=gcr.io
 time="2021-05-03T08:16:29Z" level=info msg="Setting new image to nginx:1.20.0" alias=nginx application=helm-guestbook image_name=nginx image_tag=1.20 registry=
 time="2021-05-03T08:16:29Z" level=info msg="Starting configmap/secret informers"
 time="2021-05-03T08:16:29Z" level=info msg="Successfully updated image 'nginx:1.20' to 'nginx:1.20.0', but pending spec update (dry run=false)" alias=nginx application=helm-gue
 stbook image_name=nginx image_tag=1.20 registry=
 time="2021-05-03T08:16:29Z" level=info msg="Committing 2 parameter update(s) for application helm-guestbook" application=helm-guestbook
 time="2021-05-03T08:16:29Z" level=info msg="Configmap/secret informer synced"
 time="2021-05-03T08:16:29Z" level=info msg="rm -rf /tmp/git-helm-guestbook697300132" dir= execID=9JCu3
 time="2021-05-03T08:16:29Z" level=info msg="Initializing git@github.com:jeff51419/argocd-image-updater-example.git to /tmp/git-helm-guestbook697300132"
 time="2021-05-03T08:16:29Z" level=info msg="configmap informer cancelled"
 time="2021-05-03T08:16:29Z" level=info msg="secrets informer cancelled"
 time="2021-05-03T08:16:29Z" level=warning msg="temporarily disabling strict host key checking (i.e. '-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null'), please don't
  use in production"
 time="2021-05-03T08:16:29Z" level=info msg="git fetch origin --tags --force" dir=/tmp/git-helm-guestbook697300132 execID=8tjSW
 time="2021-05-03T08:16:32Z" level=info msg="git config user.name argocd-image-updater" dir=/tmp/git-helm-guestbook697300132 execID=53OlR
 time="2021-05-03T08:16:32Z" level=info msg="git config user.email noreply@argoproj.io" dir=/tmp/git-helm-guestbook697300132 execID=ZbW1I
 time="2021-05-03T08:16:32Z" level=info msg="git checkout --force main" dir=/tmp/git-helm-guestbook697300132 execID=PRwcL
 time="2021-05-03T08:16:32Z" level=info msg="git clean -fdx" dir=/tmp/git-helm-guestbook697300132 execID=YkFpx
 time="2021-05-03T08:16:32Z" level=info msg="Successfully updated the live application spec" application=helm-guestbook
 time="2021-05-03T08:16:32Z" level=info msg="Processing results: applications=1 images_considered=2 images_skipped=0 images_updated=2 errors=0"
 time="2021-05-03T08:18:32Z" level=info msg="Starting image update cycle, considering 1 annotated application(s) for update"
 time="2021-05-03T08:18:37Z" level=info msg="Processing results: applications=1 images_considered=2 images_skipped=0 images_updated=0 errors=0"
```

**Comment by jannfis** [comment]
Hey @jeff51419, I believe the issue you're seeing has been fixed by https://github.com/argoproj-labs/argocd-image-updater/issues/170, which was released in v0.9.1 (current release is v0.9.2). Could you please update and see if your issue still persists?

**Comment by jannfis** [comment]
Possible dupe of #170

**Comment by jeff51419** [comment]
I use Kustomize from https://github.com/argoproj-labs/argocd-image-updater/manifests/base

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: argocd-image-updater
  namespace: argocd
spec:
  destination:
    namespace: argocd 
    server: 'https://kubernetes.default.svc'
  project: default
  source:
    repoURL: 'https://github.com/argoproj-labs/argocd-image-updater.git'
    path: manifests/base/
    targetRevision: HEAD
  syncPolicy:
    automated:
      prune: true
      selfHeal: true
    syncOptions:
      - CreateNamespace=true
      - ApplyOutOfSyncOnly=true
```
![image](https://user-images.githubusercontent.com/18411883/116955596-939ce580-acc5-11eb-867d-bfba0b171c50.png)

the issue still exist 
![image](https://user-images.githubusercontent.com/18411883/116955770-11f98780-acc6-11eb-83d6-cdb2e45a98c4.png)

# argocd-image-updater version
![image](https://user-images.githubusercontent.com/18411883/116955849-42412600-acc6-11eb-99ec-e8769b6b2256.png)

**Comment by jeff51419** [comment]
i also use argocd-image-updater: v0.9.2 which I still got the same issue.
![image](https://user-images.githubusercontent.com/18411883/116958892-951edb80-acce-11eb-8710-9b58004a183e.png)

There is a little difference between #170 that I assigning aliases to images (because i need to need to specify the Helm parameters used for rendering the image name and version using Helm and the parameter names do not equal image.name and image.tag)

BUT argocd-image-updater still change my image and tag to ***image.name and image.tag***

Workflow
0. annotate 
```
kubectl annotate application helm-guestbook \
argocd-image-updater.argoproj.io/write-back-method=git \
argocd-image-updater.argoproj.io/image-list="guestbook=gcr.io/heptio-images/ks-guestbook-demo, nginx=nginx" \
argocd-image-updater.argoproj.io/guestbook.helm.image-name=guestbook.image \
argocd-image-updater.argoproj.io/guestbook.helm.image-tag=guestbook.tag \
argocd-image-updater.argoproj.io/nginx.helm.image-name=nginx.image \
argocd-image-updater.argoproj.io/nginx.helm.image-tag=nginx.tag 
```
1. argocd-image-updater committed  new .argocd-source-helm-guestbook.yaml 
that is what i expected behavior
([2d7705cf647877090f8572784bd77f372bee763b](https://github.com/jeff51419/argocd-image-updater-example/commit/2d7705cf647877090f8572784bd77f372bee763b))
```yaml
helm:
  parameters:
  - name: guestbook.image
    value: gcr.io/heptio-images/ks-guestbook-demo
    forcestring: true
  - name: guestbook.tag
    value: "0.2"
    forcestring: true
  - name: nginx.image
    value: nginx
    forcestring: true
  - name: nginx.tag
    value: 1.20.0
    forcestring: true
```
2. argocd-image-updater modified .argocd-source-helm-guestbook.yaml ([6505dbe2036428d4f7039c2dc39d23ca40d4d1ee](https://github.com/jeff51419/argocd-image-updater-example/commit/6505dbe2036428d4f7039c2dc39d23ca40d4d1ee))
![image](https://user-images.githubusercontent.com/18411883/116960620-8850b680-acd3-11eb-8a45-6f9384edd211.png)
3. argocd-image-updater still modified .argocd-source-helm-guestbook.yaml image name within guestbook.image, nginx.image and  image.name. When one of the image name value change to "image.name", the image will rollback to as same as  value.yaml tag.

If i have any misunderstand, please let me know.

**Comment by jannfis** [comment]
Thanks a lot for your detailed explanation, @jeff51419! I could reproduce it successfully and a fix will be released soon.

**Comment by jannfis** [comment]
We have released v0.9.3 which contains a fix for this issue. Feel free to reopen this issue if the problem persists.

**Comment by jeff51419** [comment]
I already verified the issue been fixed.
Thanks a lot
